MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+				new_ob.location = mathutils.Vector(loc)
+				util.select_set(new_ob, True)  # needed?
+				dupedObj.append(new_ob)
+
+			# obj = new_ob # bpy.context.selected_objects[-1]
+			# do extra transformations now as necessary
+			new_ob.rotation_euler = swap.rotation_euler
+			# special case of un-applied, 90(+/- 0.01)-0-0 rotation on source (y-up conversion)
+			if (swap.rotation_euler[0]>= math.pi/2-.01 and swap.rotation_euler[0]<= math.pi/2+.01
+				and swap.rotation_euler[1]==0 and swap.rotation_euler[2]==0):
+				new_ob.rotation_euler[0] -= math.pi/2
+			new_ob.scale = swap.scale
+
+			# rotation/translation for walls, assumes last added object still selected
+			x,y,offset,rotValue,z = 0,0,0.28,0.436332,0.12
+
+			if rot == 1:
+				# torch rotation 1
+				x = -offset
+				new_ob.location += mathutils.Vector((x, y, z))
+				new_ob.rotation_euler[1]+=rotValue
+			elif rot == 2:
+				# torch rotation 2
+				y = offset
+				new_ob.location += mathutils.Vector((x, y, z))
+				new_ob.rotation_euler[0]+=rotValue
+			elif rot == 3:
+				# torch rotation 3
+				x = offset
+				new_ob.location += mathutils.Vector((x, y, z))
+				new_ob.rotation_euler[1]-=rotValue
+			elif rot == 4:
+				# torch rotation 4
+				y = -offset
+				new_ob.location += mathutils.Vector((x, y, z))
+				new_ob.rotation_euler[0]-=rotValue
+			elif rot == 5:
+				# edge block rotation 1
+				new_ob.rotation_euler[2]+= -math.pi/2
+			elif rot == 6:
+				# edge block rotation 2
+				new_ob.rotation_euler[2]+= math.pi
+			elif rot == 7:
+				# edge block rotation 3
+				new_ob.rotation_euler[2]+= math.pi/2
+			elif rot==8:
+				# edge block rotation 4 (ceiling, not 'keep same')
+				new_ob.rotation_euler[0]+= math.pi/2
+
+			# extra variance to break up regularity, e.g. for tall grass
+			# first, xy and z variance
+			if [True,1] == swapProps['variance']:
+				x = (random.random()-0.5)*0.5
+				y = (random.random()-0.5)*0.5
+				z = (random.random()/2-0.5)*0.6
+				new_ob.location += mathutils.Vector((x, y, z))
+			# now for just xy variance, base stays the same
+			elif [True,0] == swapProps['variance']: # for non-z variance
+				x = (random.random()-0.5)*0.5    # values LOWER than *1.0 make it less variable
+				y = (random.random()-0.5)*0.5
+				new_ob.location += mathutils.Vector((x, y, 0))
+
+			# Clear selection before moving on with next iteration
+			for ob in context.selected_objects:
+				if util.bv28():
+					continue
+				util.select_set(ob, False)
+		return grouped, dupedObj
+
+
+	def offsetByHalf(self, obj):
+		if obj.type != 'MESH':
+			return
+		conf.log("doing offset")
+		active = bpy.context.object #preserve current active
+		util.set_active_object(bpy.context, obj)
+		bpy.ops.object.mode_set(mode='EDIT')
+		bpy.ops.mesh.select_all(action='SELECT')
+		bpy.ops.transform.translate(value=(0.5, 0.5, 0.5))
+		bpy.ops.object.mode_set(mode='OBJECT')
+		obj.location[0] -= .5
+		obj.location[1] -= .5
+		obj.location[2] -= .5
+		util.set_active_object(bpy.context, active)
```

Comment:
(just calling out the is the endpoint of the _OT_models operator definition that I believe can be fully removed)